app/models/bulkrax/csv_entry.rb

```diff
@@ -61,7 +61,9 @@ def build_metadata
       self.parsed_metadata[Bulkrax.system_identifier_field] = [record['source_identifier']]
       record.each do |key, value|
         next if key == 'collection'
-        add_metadata(key_without_numbers(key), value)
+
+        index = key[/\d+/].to_i - 1 if key[/\d+/].to_i != 0
```

Comment:
using the csv column header to determine the index instead. the conditional is needed so that keys without numbers don't return a -1.
```
key = 'alpha_1'
index = key[/\d+/].to_i - 1 if key[/\d+/].to_i != 0        => 1

key = 'alpha'
index = key[/\d+/].to_i - 1 if key[/\d+/].to_i != 0        => nil
```